bundle/edu.gemini.lchquery.servlet/src/test/java/edu/gemini/lchquery/servlet/LchQueryFunctorTest.java

```diff
@@ -131,127 +114,45 @@ protected void addAltair(ISPProgram prog, AltairParams.Mode mode) throws SPUnkno
-     * @throws edu.gemini.pot.sp.SPTreeStateException
-     * @throws edu.gemini.pot.sp.SPNodeNotLocalException
-     */
-    protected void addTargetObsCompAO(ISPProgram prog) throws SPUnknownIDException, RemoteException, SPTreeStateException, SPNodeNotLocalException {
-        TargetObsComp target = new TargetObsComp();
-        SPTarget sptarget = new SPTarget();
-        sptarget.setName("name");
-
-        final GuideEnvironment guide = GuideEnvironment.create(
-                OptionsListImpl.<GuideGroup>create(
-                        GuideGroup.create(
-                                "group",
-                                GuideProbeTargets.create(
-                                        AltairAowfsGuider.instance,
-                                        sptarget))));
-        final TargetEnvironment env = TargetEnvironment.create(sptarget, guide, ImCollections.emptyList());
-
-        target.setTargetEnvironment(env);
-        ISPObsComponent targetObsComp = createObsComp(prog, target);
-        obscomps.add(targetObsComp);
-        obs.putClientData(IConfigBuilder.USER_OBJ_KEY, new TargetObsCompCB(targetObsComp));
-        obs.setObsComponents(obscomps);
-    }
-
-    /**
-     * Add a TargetObsComp with no targets
-     *
-     * @throws edu.gemini.pot.sp.SPUnknownIDException
-     * @throws java.rmi.RemoteException
-     * @throws edu.gemini.pot.sp.SPTreeStateException
-     * @throws edu.gemini.pot.sp.SPNodeNotLocalException
-     */
-    protected void addTargetObsCompEmpty(ISPProgram prog) throws SPUnknownIDException, RemoteException, SPTreeStateException, SPNodeNotLocalException {
-        TargetObsComp target = new TargetObsComp();
-
-        ISPObsComponent targetObsComp = createObsComp(prog, target);
-        obscomps.add(targetObsComp);
-        obs.putClientData(IConfigBuilder.USER_OBJ_KEY, new TargetObsCompCB(targetObsComp));
-        obs.setObsComponents(obscomps);
-
-
-    }
-
-    private boolean matches(ISPProgram prog) {
+    private boolean matches(final ISPProgram prog) {
         return matches(prog, 1, 1);
     }
 
-    private boolean matches(ISPProgram prog, int expectedPrograms, int expectedObservations) {
-        LchQueryFunctor functor = new LchQueryFunctor(LchQueryFunctor.QueryType.OBSERVATIONS,
-                programSemester, programTitle, programReference,
-                programActive, programCompleted, programNotifyPi, programRollover,
-                observationTooStatus, observationName, observationStatus, observationInstrument,
-                observationAo, observationClass);
-        functor.execute(null, prog, Collections.<Principal>emptySet());
-        QueryResult result = functor.getResult();
-        List<Program> programs = result.getProgramsNode().getPrograms();
+    private boolean matches(final ISPProgram prog, final int expectedPrograms, final int expectedObservations) {
+        final LchQueryFunctor functor = new LchQueryFunctor(LchQueryFunctorTestFuncs.obsQuery(),
```

Comment:
Also, there HAS to be a better way to do this than `LchQueryFunctorTestFuncs.obsQuery()`, which just returns `LchQueryFunctory.QueryType.ObservationQuery`, but I cannot for the life of me figure out how to access `LchQueryFunctory.QueryType.ObservationQuery` from Java.

Help?